src/context/authContext.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { createContext, useContext } from 'react';
+import { useUserInfoQuery } from './api/useUserInfoQuery';
+import { useNavigate } from 'react-router-dom';
+import { useQueryClient } from 'react-query';
+
+type AuthContextType = {
+  isUser: () => boolean;
```

Comment:
what does excatly `isUser` means?
I see that you check if `user.role === 'user'` so more proper name to this would be `hasUserRole`

Reply:
but what I would do instead is create function which consume user roles and find if one of the role is `user` 

something like that:

```
hasSomeOfRole = (userRoles, role) => 
   userRoles.includes(role)
```
(I don't know how roles are stored now, but if it is array of strings it should work in that way)
otherwise if user can have only one role you can just simply check `userrole === role`